cdap-source-control/src/test/java/io/cdap/cdap/sourcecontrol/operationrunner/RemoteSourceControlOperationRunnerTest.java

```diff
@@ -226,4 +231,72 @@ public void testRemoteSourceControlOperationRunnerAuthenticationConfigException(
 
     operationRunner.push(mockPushContext);
   }
+
+  @Test
+  public void testPullSuccess() throws Exception {
+    String serverURL = gitServer.getServerURL();
+    RepositoryConfig repoConfig =
+      new RepositoryConfig.Builder(mockRepoConfig).setLink(serverURL + "ignored").build();
+    PullAppContext mockPullContext = new PullAppContext(mockAppRef, repoConfig);
+
+    RemoteSourceControlOperationRunner operationRunner =
+      new RemoteSourceControlOperationRunner(cConf, metricsCollectionService, remoteClientFactory);
+    Path configFilePath = Paths.get(PATH_PREFIX, TEST_APP_NAME+".json");
+
+    addFileToGit(configFilePath, TEST_APP_SPEC, gitServer);
+
+    PullAppResponse<?> response = operationRunner.pull(mockPullContext);
+
+    // validate pull response
+    AppRequest<?> appRequest = response.getAppRequest();
+    validateTestAppRequest(appRequest);
+    Assert.assertEquals(response.getApplicationName(), mockAppDetails.getName());
+  }
+
+  @Test(expected = NotFoundException.class)
+  public void testPullNotFoundException() throws Exception {
```

Comment:
nit: can we specify what is not found in this test?

Reply:
Changed name to reflect the error scenario